src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -391,9 +391,10 @@ def _merge_input_ids_with_image_features(
         # `special_image_token_mask` identifies image tokens. Each image token will be replaced by `nb_text_tokens_per_images - 1` text tokens.
         # `torch.cumsum` computes how each image token shifts subsequent text token positions.
         # - 1 to adjust for zero-based indexing, as `cumsum` inherently increases indices by one.
-        patches_lengths = torch.tensor(patches_lengths, device=inputs_embeds.device).unsqueeze(dim=1)
-        repeated_patches = patches_lengths.repeat(1, special_image_token_mask.shape[1]) 
-        new_token_positions = torch.cumsum(special_image_token_mask * (repeated_patches - 1) + 1, -1) - 1
+        patches_lengths = torch.tensor(patches_lengths, device=inputs_embeds.device)
+        new_token_positions = torch.ones_like(input_ids)
+        new_token_positions[input_ids==self.config.image_token_index] = patches_lengths
```

Comment:
hmm but based on this seems like patches_lengths[i] is only the number of patches in a single image?

Reply:
Yes. That's what we need.